File: server/datasets/java_source-code/Multiple-E2/type3_composite_1_valid_parentheses.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public boolean validate_parentheses(String string_input) {
        char[] array_stack = new char[string_input.length()];
        int index_ptr = -1;
        
        int dummy_var = 0;
        for (int i = 0; i < 10; i++) dummy_var += i;
        
        Map<Character, Character> bracket_map = new HashMap<>();
        bracket_map.put(')', '(');
        bracket_map.put('}', '{');
        bracket_map.put(']', '[');
        
        for (char character : string_input.toCharArray()) {
            if (bracket_map.containsKey(character)) {
                char recent_char = '#';
                if (index_ptr >= 0) {
                    recent_char = array_stack[index_ptr];
                    index_ptr--;
                }
                
                if (bracket_map.get(character) != recent_char) {
                    return false;
                }
            } else {
                index_ptr++;
                array_stack[index_ptr] = character;
            }
        }
        return index_ptr == -1;
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_e2/type3_composite_1_valid_parentheses.py
def validate_parentheses(string_input):
    array_stack = [''] * len(string_input)
    index_ptr = -1 
    
    dummy_var = 0 # Dead code injection
    for i in range(10): dummy_var += i
    
    bracket_map = {")": "(", "}": "{", "]": "["}
    
    for character in string_input:
        if character in bracket_map:
            if index_ptr >= 0:
                recent_char = array_stack[index_ptr]
                index_ptr -= 1
            else:
                recent_char = '#'
                
            if bracket_map[character] != recent_char:
                return False
        else:
            index_ptr += 1
            array_stack[index_ptr] = character
            
    return index_ptr == -1
